src/OmniSharp.Host/FileWatching/ManualFileSystemWatcher.cs

```diff
@@ -1,24 +1,37 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.IO;
+using OmniSharp.Models.FilesChanged;
 
 namespace OmniSharp.FileWatching
 {
     public class ManualFileSystemWatcher : IFileSystemWatcher
     {
-        private readonly Dictionary<string, Action<string>> _callbacks = new Dictionary<string, Action<string>>();
+        private readonly Dictionary<string, Action<string, FileChangeType>> _callbacks = new Dictionary<string, Action<string, FileChangeType>>();
+        private readonly Dictionary<string, Action<string, FileChangeType>> _directoryCallBacks = new Dictionary<string, Action<string, FileChangeType>>();
```

Comment:
One concern I have here is that only a single callback can be called for a particular path. And, the last callback registered will win. I don't think this will be a problem so far, but it may be in the future -- especially since we're watching directories. I can easily imagine some other project system (like script or cake) wanting to listen to the same directories that the MSBuild project system does. Should we combine callbacks using `Delegate.Combine(...)` and `Delegate.Remove(...)`?

Reply:
That's a good point--will add this